plugin/gcp_config.go

```diff
@@ -61,16 +61,24 @@ func (c *gcpConfig) Update(d *framework.FieldData) error {
 	}
 
 	if v, ok := d.GetOk("credentials"); ok {
-		creds, err := gcputil.Credentials(v.(string))
-		if err != nil {
-			return fmt.Errorf("failed to read credentials: %w", err)
-		}
+		credentials := v.(string)
+
+		// If the given credentials are empty, reset them so that application default
+		// credentials are used. Otherwise, parse and validate the given credentials.
+		if credentials == "" {
+			c.Credentials = nil
+		} else {
```

Comment:
Overall looks good to me.

I have a question about the logic this is based on, if you know. It looks to me like [this line](https://github.com/hashicorp/vault-plugin-auth-gcp/blob/a6717f99edfbe8c64517529a7a66f1f68962ce4a/plugin/backend.go#L213=) is where it's deciding whether to use Application Default Credentials or the credentials provided inside of the config. How come overriding credentials to `""`, before your PR, does not satisfy `len(credBytes)` to be 0 and cause it to enter the else block?

Reply:
Thanks for the explanation, I see it now.